src/ucp/core/ucp_mm.c

```diff
@@ -351,41 +352,71 @@ static void ucp_memh_dereg(ucp_context_h context, ucp_mem_h memh,
             continue;
         }
 
-        ucs_trace("de-registering memh[%d]=%p", md_index, memh->uct[md_index]);
-        ucs_assert(context->tl_mds[md_index].attr.flags & UCT_MD_FLAG_REG);
-
         params.memh = memh->uct[md_index];
         if (memh->inv_md_map & UCS_BIT(md_index)) {
             params.flags = UCT_MD_MEM_DEREG_FLAG_INVALIDATE;
+            if (rkey_only) {
+                params.flags |= UCT_MD_MEM_DEREG_FLAG_INVALIDATE_RKEY_ONLY;
+            }
```

Comment:
you can pass `uct_flags` instead of `rkey_only` and pass `UCT_MD_MEM_DEREG_FLAG_INVALIDATE_RKEY_ONLY` directly. This would simplify this code

Reply:
refactored